addons/pos_self_order/models/res_config_settings.py

```diff
@@ -97,32 +102,58 @@ def _generate_single_qr_code(self, url):
             'svg': qr.make_image(fill_color="black", back_color="transparent", image_factory=qrcode.image.svg.SvgImage)
         }
 
+    def _generate_excel(self, rows, headers):
+        buffer = BytesIO()
+        workbook = xlsxwriter.Workbook(buffer, {'in_memory': True})
+        worksheet = workbook.add_worksheet()
+
+        for col, header in enumerate(headers):
+            worksheet.write(0, col, header)
+        for row_idx, row in enumerate(rows, start=1):
+            for col_idx, cell in enumerate(row):
+                worksheet.write(row_idx, col_idx, cell)
+
+        workbook.close()
+        return buffer.getvalue()
```

Comment:
Is it possible to do something like:

```suggestion
    with BytesIO() as buffer:
        with xlsxwriter.Workbook(buffer, {'in_memory': True}) as workbook:
            worksheet = workbook.add_worksheet()

            for col, header in enumerate(headers):
                worksheet.write(0, col, header)
            for row_idx, row in enumerate(rows, start=1):
                for col_idx, cell in enumerate(row):
                    worksheet.write(row_idx, col_idx, cell)
        return buffer.getvalue()
```

So that you would not need to care about opening and closing the workbook and the buffer, it is done automatically